### PR TITLE
Fall2017

### LINKED ISSUE #238: Improve Rel Preload Approach to be standalone
The main priority of the loadCSS project is to provide a simple workflow around loading CSS with Rel=preload, so our existing cssrelpreload.js is in need of some updates.

1. The cssrelpreload.js file should not need to depend upon loadCSS or onloadCSS, as it can reuse and manipulate link elements that are already in the page.  This should simplify and lighten the scripting overall if you're only using loadCSS for a rel=preload helper.
2. It would be nice if we no longer use cookies and instead test alternative methods again, like localStorage, for flagging a stylesheet as already-in-cache

The work for this refactor is happening in the https://github.com/filamentgroup/loadCSS/tree/fall2017 branch

### PR DESCRIPTION
**Pull request by scottjehl**
Refactor rel=preload workflow to be a standalone JS file (no longer dependent upon loadCSS).

Reorganize test files and explain them a bit as well
fixes #238

### CODE REVIEW

File: src/cssrelpreload.js [hunk 1]
```diff
@@ -13,32 +20,66 @@
...
-    }
+  rp.poly();
+  var run = w.setInterval( rp.poly, 300 );
+  if( w.addEventListener ){
+    w.addEventListener( "load", function(){
+      rp.poly();
+      w.clearInterval( run );
+    } );
+  }
+  if( w.attachEvent ){
```
Comment (odahcam) [inline src/cssrelpreload.js]: I think that there's should be an `else` here, cause in browsers that support both `addEventListener` and `attachEvent`, the listener will be registered twice.

Comment (scottjehl) [reply src/cssrelpreload.js]: Thanks @odahcam I'd say that's a good idea.

**Comment by odahcam** [comment]
@scottjehl also we could add that minified version people are asking for, once for all. 😃

**Comment by MarcoHengstenberg** [comment]
❤️ Thanks for this!

File: src/cssrelpreload.js [hunk 2]
```diff
@@ -13,32 +21,59 @@
...
         w.clearInterval( run );
       } );
     }
-  }
-}( this ));
+  };
+  // run once at least
+  rp.poly();
+  // commonjs
+	if( typeof exports !== "undefined" ){
```
Comment (zachleat) [inline src/cssrelpreload.js]: using tabs here

**Pull request merged**